qiskit_metal/analyses/base/analysis_interfaces.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+class NeedsRenderer():
+    """Just a way to centralize the selection and naming of the renderer.
+    Could refactor code to transform this class into a class-less function.
+    """
+    def __init__(self, design: 'QDesign', renderer_name: str, *args, **kwargs):
+        """Variables and method needed from all those Analysis types that need a renderer.
+
+        Args:
+            design (QDesign): The Metal design you are working on.
+            renderer_name (str): A string corresponding to the name of the renderer you intend to use.
+        """
+        super().__init__(*args, **kwargs)
```

Comment:
The super() isn't doing anything given no class is being inherited right?

Reply:
It does inherit implicitly from `object`. And `object` is supposed to be called with no __init__ arguments, otherwise it means that more arguments were passed then expected. There is other ways to do that without the use of super.
https://stackoverflow.com/questions/52272811/when-inheriting-directly-from-object-should-i-call-super-init

There is another reason I have done that, which is related to multiple inheritance. I do not recall now but without that line only one of the two classes seemed to run. I might be wrong on the reasons and I will investigate this further.

Reply:
I confirmed that we need that for multiple class inheritance.